### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by myaserkhan**
**For the third milestone of the OPP School Library project covered the following requirements**

- Create a class Classroom with @label instance variable that should be initialized in the constructor and add Setter and getter for @label
- Create the has-many/belongs-to relationship between Classroom and Student.
- Create a class Book with @title and @author instance variables that should be initialized in the constructor and add Setters and getters for instance variables.
- Create a class Rental with @date instance variable that should be initialized in the constructor and add Setter and getter for @date
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental.

### CODE REVIEW

File: Relationship/book.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.book = self
+  end
```
Comment (ha-manel) [inline Relationship/book.rb]: - [ ] Kindly create a `rental` in this method, this method should accept `person` and `date` as arguments and inside it should create a `rental` object using `self` and the arguments like this:

```
Rental.new(date, self, person)
```

File: Entities/person.rb [hunk 2]
```diff
@@ -27,4 +29,9 @@ def can_use_services?
   def correct_name
     @name
   end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.person = self
+  end
```
Comment (ha-manel) [inline Entities/person.rb]: - [ ] As mentioned earlier, kindly create a rental object using this method, it should have `date` and `book` as arguments and inside it should create an instance like this:

```
Rental.new(date, book, self)
```

**Review by ha-manel — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️

## Hi @myaserkhan 👋 
Great job so far, There are some changes that need to be implemented in order to proceed to the next project 👍  



## Highlights:  
* All tests passed ✔️
* Meaningful commit messages ✔️ 
* Professional README file ✔️

## Required Changes:

- [ ] Kindly add the MIT file to your project because the link in your README file is broken👍 

_Kindly check the comments below_

------
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Nemwel-Boniface — APPROVED** [review]
## Hello @myaserkhan           :wave: ,
I am Nemwel your reviewer for this PR :man_beard:  

Thank you for meeting all the project requirements good job! :partying_face: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :partying_face: 🎉 



## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 
Met all the milestone requirements ✔️ 

## [Optional] suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**